CIP-AFFIRM/README.md

```diff
@@ -0,0 +1,126 @@
+
+If we are to bank the unbanked, we should also provide ID for those who have none.
+
+In this CIP we propose a simple, truly decentralized mechanism for identity verification and trust.
+
+### Why does this require a CIP? 
+We are aiming to create a scheme which is generic and becomes the accepted standard for registering your affirmation of a particular wallet, such that everyone may participate and we can be sure that there is one definitive affirmation graph for the blockchain and that all affirmation transactions will follow a consistent format. As such, a defined standard that everyone can follow is necessary - a CIP is the way to acheive this. 
+
+To that end, a smart contract conforming to this specification has been defined which attempts to be the most simple and generic possible.  No fee is required (aside from the inherent blockchain fees), the smart contract contains nothing specific to the author's websites or implementation. It is designed so that it can be integrated into any blockchain explorer, wallet or dApp. It is expected that the majority will use this generic smart contract, but it's also possible to define your own smart contract that is compliant with this CIP - it must meet the specific requirements detailed below.
+
+## Specification
+[v1 of the affirmation contract](https://github.com/kieransimkin/affirmation-graph/blob/main/validators/affirmation.ak)  
+
+Definitions:  
+`target` - the recipient of the affirmation  
+`source` - the creator of the affirmation (the person or entity who is vouching for them)  
+`target hash` - if the `target` is a normal wallet, this should be their stake key hash. If the `target` is a script, this should be the script hash  
+`source hash` - if the `source` is a normal wallet, this should be their stake key hash. If the `source` is a script, this should be the script hash  
+`affirmation contract` - any smart contract which conforms to this specification (details below)  
+`contract hash` - the hash of the `affirmation contract`  
+`affirmation address` - An address where the spending part is `contract hash` and the delegation part is `target hash`  
+
+An affirmation consists of a UTxO at the `affirmation address` with an inline datum containing the `source hash`. To be valid, this UTxO must contain one token minted by the `affirmation contract`, and the `affirmation contract` itself must be valid as defined below:
```

Comment:
> the script does also require minting a kinda pointless token - but this is just a necessary trick to get Cardano to perform validation upon creation of the affirmation UTxO, since Cardano does not otherwise perform validation upon creating a UTxO - I wonder, is there a CIP for that?

@fallen-icarus this sounds familiar & I think you've covered that realm... would you recommend any approach either from a merged or candidate CIP?